test/distributed/org/apache/cassandra/fuzz/harry/integration/dsl/HistoryBuilderOverridesTest.java

```diff
@@ -0,0 +1,376 @@
+                                                             ColumnSpec.staticColumn("static2", ColumnSpec.int64Type)
+                                               ));
+
+            Map<String, Bijections.Bijection<?>> reGenerators = new HashMap<>();
+            reGenerators.put("regular1", ColumnSpec.asciiType(4, 4).generator());
+            reGenerators.put("regular2", ColumnSpec.int64Type.generator());
+            reGenerators.put("static1", ColumnSpec.asciiType(8, 4).generator());
+            reGenerators.put("static2", ColumnSpec.int64Type.generator());
+
+            DataTracker tracker = new DefaultDataTracker();
+            beforeEach();
+            sut.schemaChange(schema.compile().cql());
+
+            TokenPlacementModel.ReplicationFactor rf = new TokenPlacementModel.SimpleReplicationFactor(1);
+
+            int partitionSize = 100;
+            HistoryBuilder history = new HistoryBuilder(seed, partitionSize, 10, schema, rf);
+            ReplayingVisitor visitor = history.visitor(tracker, sut, SystemUnderTest.ConsistencyLevel.ALL);
+            EntropySource rng = new JdkRandomEntropySource(seed);
+
+            Map<String, Set<Object>> perColumnOverrides = new HashMap<>();
+            for (ColumnSpec column : schema.regularColumns)
+            {
+                perColumnOverrides.put(column.name, new HashSet<>());
+                for (int i = 0; i < partitionSize; i++)
+                {
+                    Object override = reGenerators.get(column.name).inflate(rng.next());
+                    history.valueOverrides().override(column.name, i, override);
+                    perColumnOverrides.get(column.name).add(override);
+                }
+            }
+
+            for (ColumnSpec column : schema.staticColumns)
+            {
+                perColumnOverrides.put(column.name, new HashSet<>());
+                for (int i = 0; i < partitionSize; i++)
+                {
+                    Object override = reGenerators.get(column.name).inflate(rng.next());
+                    history.valueOverrides().override(column.name, i, override);
+                    perColumnOverrides.get(column.name).add(override);
+                }
+            }
+            for (int pdIdx = 0; pdIdx < 10; pdIdx++)
+            {
+                Map<String, Set<Object>> results = new HashMap<>();
+                for (ColumnSpec column : schema.regularColumns)
+                    results.put(column.name, new HashSet<>());
+                for (ColumnSpec column : schema.staticColumns)
+                    results.put(column.name, new HashSet<>());
+
+                for (int i = 0; i < partitionSize; i++)
+                {
+                    history.visitPartition(pdIdx)
+                           .insert(i,
+                                   new int[]{ rng.nextInt(100), rng.nextInt(100) },
+                                   new int[]{ rng.nextInt(100), rng.nextInt(100) });
```

Comment:
I think I made a comment about how we name these in the `SingleOperationBuilder` interface, but to confirm, we're basically asking the history builder to insert our overrides at the random row indexes?

Reply:
Right: we are simply saying "insert a value number X", where X is generated randomly. There's nothing preventing us from putting `210` as an index, it's just for the purpose of our test we ensure that _all_ the values were overriden. You can still generate arbitrary random values from the indexes you have not overriden.